pycicle_params.py

```diff
@@ -0,0 +1,79 @@
+#  Copyright (c) 2017-2018 John Biddiscombe, Peter Doak
+#
+#  Distributed under the Boost Software License, Version 1.0. (See accompanying
+#  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+#--------------------------------------------------------------------------
+import os
+import re
+
+class PycicleParamsHelper:
+    @staticmethod
+    def no_op(self, *args, **kwargs):
+        pass
+
+class PycicleParams:
+    keys = ['PYCICLE_PROJECT_NAME',
+            'PYCICLE_GITHUB_PROJECT_NAME',
+            'PYCICLE_GITHUB_ORGANISATION',
+            'PYCICLE_PR',
+            'PYCICLE_BRANCH',
+            'PYCICLE_MASTER',
+            'PYCICLE_HOST',
+            'PYCICLE_ROOT',
+            'PYCICLE_RANDOM',
+            'PYCICLE_COMPILER_TYPE',
+            'PYCICLE_BOOST',
+            'PYCICLE_BUILD_TYPE',
+            'PYCICLE_JOB_SCRIPT_TEMPLATE',
+            'PYCICLE_SRC_ROOT',
+            'PYCICLE_BUILD_ROOT',
+            'PYCICLE_LOCAL_GIT_COPY',
+            'PYCICLE_PR_ROOT',
+            'PYCICLE_BINARY_DIRECTORY',
+            'PYCICLE_PR_',
+            'PYCICLE_CTEST_BUILD_TARGET',
+            'PYCICLE_GITHUB_TOKEN',
+            'PYCICLE_MACHINES',
+            'PYCICLE_MACHINE',
+            'PYCICLE_HTTP',
+            'PYCICLE_JOB_LAUNCH',
+            'PYCICLE_GITHUB_MASTER_BRANCH',
+            'PYCICLE_CDASH_SERVER_NAME',
+            'PYCICLE_CDASH_PROJECT_NAME',
+            'PYCICLE_CDASH_HTTP_PATH',
+            'PYCICLE_BUILD_STAMP',
+            'PYCICLE_COMPILER_SETUP',
+            'PYCICLE_SLURM',
```

Comment:
same comment here - see refactoroed slurm/pbs option

Reply:
Yes part of the point here is to support some config error catching.